File: LeetCode/June Challenge/11-Sort Colors/solution.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        if(nums.size()==0 || nums.size()==1)
            return;
        
        int start=0,end=nums.size()-1,index=0;
        
        while(index<=end && start<end)
        {
            if(nums[index]==0)
            {
                swap(nums[index],nums[start]);
                start++;
                index++;
            }
            else if(nums[index]==2)
            {
                swap(nums[index],nums[end]);
                end--;
            }
            else
                index++;
        }
    }
};

File: LeetCode/June Challenge/11-Sort Colors/solution.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        l = curr = 0
        r = len(nums) - 1
        
        while curr <= r:
            if nums[curr] == 2:
                nums[curr], nums[r] = nums[r], nums[curr]
                r -= 1
            elif nums[curr] == 0:
                nums[l], nums[curr] = nums[curr], nums[l]
                l += 1
                curr += 1
            else:
                curr += 1
